### PR TITLE
Create beatmaps service layer with jit logic; track bancho ranked status & updated_at separately

### PR DESCRIPTION
**Pull request by cmyui**

### CODE REVIEW

File: server/main.py [hunk 1]
```diff
@@ -681,47 +681,10 @@ async def get_scores_handler(
     ):
         return
 
-    beatmap = await beatmaps.fetch_one_by_md5(beatmap_md5)
```
Comment (cmyui) [inline server/main.py:685]: this logic has been moved to the services layer

File: server/services/beatmaps.py [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+        "creator": osu_api_beatmap.beatmapset.creator_name,
+        "filename": _create_beatmap_filename(
+            artist=osu_api_beatmap.beatmapset.artist,
+            title=osu_api_beatmap.beatmapset.title,
+            version=osu_api_beatmap.version,
+            creator=osu_api_beatmap.beatmapset.creator_name,
+        ),
+        "total_length": osu_api_beatmap.total_length,
+        "max_combo": osu_api_beatmap.max_combo or 0,
+        "ranked_status_manually_changed": False,  # manually ranked
```
Comment (tsunyoku) [inline server/services/beatmaps.py]: if this is being tracked, may it be a good idea to track bancho's status vs ours?

Comment (cmyui) [reply server/services/beatmaps.py]: yeah you're right, also i think we want to do the same for updated_at

both bancho's and ours

Comment (tsunyoku) [reply server/services/beatmaps.py]: oh yeah definitely, i missed out that one

**Pull request merged**